File: src/Shared/Logistics.Shared.Models/SubscriptionDto.cs
using Logistics.Shared.Enums;

namespace Logistics.Shared.Models;

public class SubscriptionDto
{
    public required string Id { get; set; }
    public SubscriptionStatus Status { get; set; }
    public required TenantDto Tenant { get; set; }
    public required SubscriptionPlanDto Plan { get; set; }
    public DateTime StartDate { get; set; } = DateTime.UtcNow;
    public DateTime? EndDate { get; set; }
    public DateTime? NextPaymentDate { get; set; }
    public DateTime? TrialEndDate { get; set; }
}

File: src/Client/Logistics.OfficeApp/src/app/core/models/subscription/subscription.dto.ts
import {SubscriptionStatus} from "./enums";
import {SubscriptionPlanDto} from "./subscription-plan.dto";

export interface SubscriptionDto {
  id: string;
  status: SubscriptionStatus;
  plan?: SubscriptionPlanDto;
  startDate: Date;
  endDate?: Date;
  nextPaymentDate?: Date;
  trialEndDate?: Date;
  stripeSubscriptionId?: string;
  stripeCustomerId?: string;
}
